src/test/java/org/reaktivity/nukleus/maven/plugin/internal/generated/FlatFWTest.java

```diff
@@ -61,57 +72,136 @@ public void shouldProvideTypeId() throws Exception
+
+        assertEquals(expectedLimit, limit);
+        assertEquals(expected, buffer);
+    }
+
+    @Test
+    public void shouldSetAllValues2() throws Exception
+    {
+        final int offset = 0;
+        int expectedLimit = setAllBufferValues(expected, offset);
+
+        Builder builder = setAllFieldsValue2(flatRW.wrap(buffer, offset, expectedLimit));
+
+        int limit = builder.build().limit();
+
+        assertEquals(expectedLimit, limit);
+        assertEquals(expected, buffer);
+    }
+
+    @Test
+    public void shouldReadDefaultedValues() throws Exception
+    {
+        final int offset = 11;
+        int limit = setAllRequiredBufferValues(buffer, offset);
+        assertSame(flatRO, flatRO.wrap(buffer,  offset,  limit));
+        assertRequiredValuesAndDefaults(flatRO);
+    }
+
+    @Test
+    public void shouldSetStringValuesToEmptyString() throws Exception
+    {
+        int limit = flatRW.wrap(buffer, 0, buffer.capacity())
                 .fixed1(10)
-                .string1("value1")
-                .string2("value2")
+                .fixed2(20)
+                .string1("")
+                .fixed3(30)
+                .string2("")
                 .build()
                 .limit();
         flatRO.wrap(buffer,  0,  limit);
-        assertEquals(222, flatRO.fixed2());
-        assertEquals(333, flatRO.fixed3());
+        assertEquals("", flatRO.string1().asString());
+        assertEquals("", flatRO.string2().asString());
     }
 
     @Test(expected = IndexOutOfBoundsException.class)
     public void shouldFailToSetFixed1WithInsufficientSpace()
     {
         flatRW.wrap(buffer, 10, 10)
-               .fixed1(10);
+                .fixed1(10);
     }
 
-    @Test(expected = IndexOutOfBoundsException.class)
-    public void shouldFailToSetFixed2WithInsufficientSpace()
+    @Test
+    public void shouldFailToResetFixed1() throws Exception
     {
-        flatRW.wrap(buffer, 10, 12)
-                .fixed1(10)
-                .fixed2(20);
+        expectedException.expect(IllegalStateException.class);
+        expectedException.expectMessage("fixed1");
+        setRequiredFields(flatRW.wrap(buffer, 0, 100), INDEX_FIXED2)
+                .fixed1(101)
+                .build();
     }
 
-    @Test(expected = IndexOutOfBoundsException.class)
-    public void shouldFailToSetString1WhenDefaultingFixed2ExceedsMaxLimit()
+    @Test
+    public void shouldFailToBuildWhenFixed1NotSet() throws Exception
     {
-        flatRW.wrap(buffer, 10, 12)
-                .fixed1(10)
-                .string1("");
+        expectedException.expect(IllegalStateException.class);
+        expectedException.expectMessage("fixed1");
+        flatRW.wrap(buffer, 0, 100)
+                .build();
     }
 
-    @Test(expected = IndexOutOfBoundsException.class)
-    public void shouldFailToSetString1WhenExceedsMaxLimit()
+    @Test(expected = IllegalArgumentException.class)
+    public void shouldFailToSetFixed1WithLowerValue()
```

Comment:
shouldFailToSetFixed1WithValueTooLow (same for the other methods)

Reply:
Done.